### PR TITLE
hack: add verification of go deps in gh action

### PR DESCRIPTION
**Pull request by rewantsoni**
update Make file to add new targets to verify go deps and add it in the github action

### CODE REVIEW

File: .github/workflows/verify-generated.yaml [hunk 1]
```diff
@@ -17,13 +17,20 @@ on:
...
       - uses: actions/checkout@v4
 
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
       - name: Verify changes to CSI images manifest
         run: make verify-csi-images-manifest
+
+      - name: Verify go deps
```
Comment (leelavg) [inline .github/workflows/verify-generated.yaml:35]: just a thought, seeing the actions log it seems we are going to download deps everytime and when you run the same in your machine it'll get copied from global cache afaik.

do you think it's worth to cache the dependencies as well?

Comment (leelavg) [reply .github/workflows/verify-generated.yaml:35]: disregard, I just [saw](https://github.com/actions/setup-go?tab=readme-ov-file#caching-dependency-files-and-build-outputs) cache is on by default

**Comment by rchikatw** [comment]
LGTM

**Review by leelavg — APPROVED** [review]

**Pull request merged**